ihatemoney/forms.py

```diff
@@ -110,6 +111,10 @@ class EditProjectForm(FlaskForm):
     default_currency = SelectField(_("Default Currency"), validators=[DataRequired()])
 
     def __init__(self, *args, **kwargs):
+        if not hasattr(self, "id"):
+            # We must access the project to validate the default currency.
+            # Mimics a StringField. Defaut to empty string to ensure that query run smoothly.
+            self.id = SimpleNamespace(data=kwargs.pop("id", ""))
```

Comment:
Instead of adding an ID, you should be able to query project by name:

```
models.Project.query.get_by_name(name=XXX)
```

Reply:
Thanks, but name are not unique in database, so we need to discriminate them.

Reply:
Ok, that makes sense.

Reply:
So, I guess we do need this hack.

I had a hard time understanding the code (which led me to open #774 ), so please add more comments ^^ For instance on why you need the `not hasattr(self, "id")` and what is this `SimpleNamespace` thing.

Reply:
OK, done. I didn't explain the SimpleNamespace behavior, since it's in the default library.

Reply:
Thanks!